plugins/catalog/src/components/ComponentRemovalDialog/ComponentRemovalDialog.tsx

```diff
@@ -101,10 +104,25 @@ export const ComponentRemovalDialog: FC<ComponentRemovalDialogProps> = ({
         ) : null}
       </DialogContent>
       <DialogActions>
-        <Button onClick={onClose}>Cancel</Button>
+        <Button onClick={onClose} color="primary">
+          Cancel
+        </Button>
         <Button
           disabled={!!(loading || error)}
-          onClick={onConfirm}
+          onClick={async () => {
+            const uid = entity.metadata?.uid;
+            if (uid) {
```

Comment:
uid will always be present for things fetched from the api, just do `await catalogApi.removeEntityByUid(uid!)` if needed